artifactory/commands/oc/startbuild.go

```diff
@@ -0,0 +1,193 @@
+		return err
+	}
+	log.Info("Build", ocBuildName, "finished successfully.")
+
+	buildName := osb.buildConfiguration.BuildName
+	buildNumber := osb.buildConfiguration.BuildNumber
+	project := osb.buildConfiguration.Project
+
+	saveBuildInfo := buildName != "" && buildNumber != ""
+	if !saveBuildInfo {
+		return nil
+	}
+
+	log.Info("Collecting build info...")
+	// Get the new image details from OpenShift
+	imageTag, manifestSha256, err := oc.GetImageDetails(osb.executablePath, ocBuildName)
+	if err != nil {
+		return err
+	}
+
+	if err := utils.SaveBuildGeneralDetails(buildName, buildNumber, project); err != nil {
+		return err
+	}
+	serviceManager, err := utils.CreateServiceManagerWithThreads(osb.serverDetails, false, osb.threads, -1)
+	if err != nil {
+		return err
+	}
+	image := container.NewImage(imageTag)
+	builder, err := container.NewBuildInfoBuilderForKanikoOrOpenShift(image, osb.repo, buildName, buildNumber, project, serviceManager, container.Push, manifestSha256)
+	if err != nil {
+		return err
+	}
+	buildInfo, err := builder.Build(osb.buildConfiguration.Module)
+	if err != nil {
+		return err
+	}
+
+	log.Info(fmt.Sprintf("OpenShift CLI (oc) start-build command finished successfully."))
+	return utils.SaveBuildInfo(buildName, buildNumber, project, buildInfo)
+}
+
+func (osb *OcStartBuildCommand) ServerDetails() (*config.ServerDetails, error) {
+	return osb.serverDetails, nil
+}
+
+func (osb *OcStartBuildCommand) CommandName() string {
+	return "rt_oc_start_build"
+}
+
+func (osb *OcStartBuildCommand) setOcExecutable() error {
+	ocExecPath, err := exec.LookPath("oc")
+	if err != nil {
+		return errorutils.CheckError(err)
+	}
+
+	osb.executablePath = ocExecPath
+	log.Debug("Found OpenShift CLI executable at:", osb.executablePath)
+	return nil
+}
+
+func (osb *OcStartBuildCommand) SetArgs(args []string) *OcStartBuildCommand {
+	osb.ocArgs = args
+	return osb
+}
+
+func (osb *OcStartBuildCommand) validateAllowedOptions() error {
+	notAllowedOcOptions := []string{"-w", "--wait", "--template", "-o", "--output"}
+	for _, arg := range osb.ocArgs {
+		for _, optionName := range notAllowedOcOptions {
+			if arg == optionName || strings.HasPrefix(arg, optionName+"=") {
+				return errorutils.CheckError(errors.New(fmt.Sprintf("the %s option is not allowed", optionName)))
+			}
+		}
+	}
+	return nil
+}
+
+func (osb *OcStartBuildCommand) validateOcVersion() error {
+	ocVersionStr, err := oc.Version(osb.executablePath)
+	if err != nil {
+		return err
+	}
+	trimmedVersion := strings.TrimPrefix(ocVersionStr, "v")
+	ocVersion := version.NewVersion(trimmedVersion)
+	if ocVersion.Compare(minSupportedOcVersion) > 0 {
+		return errorutils.CheckError(errors.New(fmt.Sprintf(
+			"JFrog CLI oc-start-build command requires OpenShift CLI version " + minSupportedOcVersion + " or higher")))
+	}
+	return nil
+}
+
+func extractOcOptionsFromArgs(args []string) (repo, serverId string, threads int, buildConfig *utils.BuildConfiguration, cleanArgs []string, err error) {
```

Comment:
It makes more sense for the flags extraction, including the validation to be done in jfrog-cli and not in jfrog-cli-core.

Reply:
I moved the extraction part to jfrog-cli, but I didn't move the validation part.
It validates the args passed to oc only, and not all of the args of the command (like repo, build-name, etc).